src/platform/Darwin/PlatformManagerImpl.cpp

```diff
@@ -53,6 +53,27 @@ CHIP_ERROR PlatformManagerImpl::_InitChipStack()
     return err;
 }
 
+CHIP_ERROR PlatformManagerImpl::_StartEventLoopTask()
+{
+    dispatch_resume(mWorkQueue);
```

Comment:
The docs for `dispatch_resume` say that calling it when the suspension count is 0 is undefined behavior.

We have zero docs for `RunEventLoop` and `StartEventLoopTask` so I am still sorting out their semantics.  But does anything prevent a consumer from calling both?  And if they do, won't we end up with unbalanced counts?  Do we need to have our own counter to handle this?

Also, it seems like the semantics of `RunEventLoop` are "Keep spinning until StopEventLoopTask" happens or so, no?  If that's the case, doing `dispatch_resume` here seems a little weird....

Reply:
`RunEventLoop` should not be called alongside `StartEventLoopTask` - they have different purposes.

The former is called when you don't want to spin up a separate threading context for the stack and want to block on processing events on the stack's dispatch queue.

The latter is called when you want to spin up a separate threading context for the stack.

Reply:
> The docs for `dispatch_resume` say that calling it when the suspension count is 0 is undefined behavior.
> 

> We have zero docs for `RunEventLoop` and `StartEventLoopTask` so I am still sorting out their semantics. But does anything prevent a consumer from calling both? And if they do, won't we end up with unbalanced counts? Do we need to have our own counter to handle this?

I sort of fixes that by adding a boolean to ensure we are not getting out of sync here.

None of the example are calling both as far as I can tell. 

> 
> Also, it seems like the semantics of `RunEventLoop` are "Keep spinning until StopEventLoopTask" happens or so, no? If that's the case, doing `dispatch_resume` here seems a little weird....

That's a bit more complicated than this as far as I can tell. `RunEventLoop` basically start processing events on whatever caller thread, so it is effectively starting to process events. This is why this PR does `dispatch_resume` here, in order to start processing events.